File: tyama_icpc2007rA.lua
#!/usr/bin/lua
while true do
	n=io.read("*n")
	k=io.read("*n")
	m=io.read("*n")
	if n==0 then break end
	r=0
	for i=1,n-1 do r=(r+k)%i end
	print((r+m)%n+1)
end

File: tyama_icpc2007rA.clj
#!/usr/bin/clojure

(defn calc [r i n k]
	(if (= i n) r (recur (rem (+ r k) i) (+ i 1) n k))
)

(defn ma []
	(let [n (read) k (read) m (read)]
		(if (not (zero? n)) (do
			(println (+ (rem (+ (calc 0 1 n k) m) n) 1))
			(recur)
		))
	)
)
(ma)